### PR TITLE
Schema Descriptors Framework

### PR DESCRIPTION
**Pull request by trieloff**
The actual schemas will come in separate PRs, imported from #154 

I'm splitting the big PR into multiple smaller PRs, so that we can discuss every schema descriptor on its own merits.

### CODE REVIEW

**Comment by trieloff** [comment]
@lrosenthol - last chance to object…

**Comment by lrosenthol** [comment]
@treiloff if you are just asking if I am OK with splitting it up and we will review each of the smaller ones individually - that’s fine.   Anything else, no comment

**Comment by trieloff** [comment]
@lrosenthol then please review this PR. @kstreeter is fine with it (he wrote it) and I'm fine with it, too (I do have problems with some of the descriptors that will come in later PRs)

File: docs/descriptors.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+* `xdm:oneToMany`: describes a 1:m relationship between a source schema and a destination schema
+* `xdm:manyToMany`: describes an m:n relationship between a source schema and a destination schema
+
+These relationships are defined in XDM using the `RelationshipDescriptor` schema.
+
+## Update Policies
+
+Data described by an XDM schema may change over time, and as such a data object may reflect an update of a previous instance of that object. There are different ways that an update may be handled, and this way depends both on the nature of the data and the specific application it is being used for.
+
+XDM defines a schema descriptor of type `xdm:descriptorUpdatePolicy`, which describes several common methods of handling an update:
```
[the remark this answers is not in the document]
Comment (kstreeter) [reply docs/descriptors.md]: Yes, it is about how the data described by the schema is updated.  How the update occurs can be described on a property-by-property basis.  The update policies might be different for different datasets that share a schema, which is why we need to put this in a descriptor versus just annotating the schema in the repo.

[the remark this answers is not in the document]
Comment (lrosenthol) [reply docs/descriptors.md]: but isn't this outside the scope of XDM and the schemas?  This seems like a business thing, perhaps even specific to XC?

File: docs/descriptors.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+          "xdm:property": "code"
+        }
+      ],
+      "type": "string"
+    }
+  }
+}
+```
+
+The customer ID is present, but does not contain other information needed to ensure the identity is fully described, such as the ID namespace, or whether this value represents the application's native ID for this customer or if this is an ID given my some external system.
```
[the remark this answers is not in the document]
Comment (lrosenthol) [reply docs/descriptors.md]: so just as with the other case, this seems very specific to your business and not to XDM in general

[the remark this answers is not in the document]
Comment (kstreeter) [reply docs/descriptors.md]: The concept of end user identity is a key aspect of both how XDM models data about end-user interactions (`ExperienceEvents`) and how that is translated to an actionable view of the end user (`Profile`).  The `Identity` and `Namespace` schemas describe the framework used to manage that end user identity.  All of the schemas mentioned are part of XDM.  

This descriptor solves a part of that overall problem: sometimes we need to connect data modeled as XDM (like EE and Profile) back to data that isn't modeled in XDM (for example, a legacy customer management system).  It is likely that anyone using EE or Profile would have this problem, and need this descriptor.

[the remark this answers is not in the document]
Comment (lrosenthol) [reply docs/descriptors.md]: Yes, and EEs are very tied to your business - and that's fine.  But they have nothing to do with mine or that of CC.   As such, identity is not a core piece of our XDM strategy.  
I am perfectly fine with this being an XC-specific extension to XDM - but as is, it is not something that belongs in the "core" (IMO).

File: docs/descriptors.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+* `xdm:updateReplace`: the new data object should replace the existing data object
+* `xdm:updateTimeSeries`: the data is time series data, and the new object should be logged/collected without changing any existing data
+
+Update policies are defined using the `UpdatePolicyDescriptor` schema.
+
+## Other Supported Schema Descriptors
+
+A number of additional schema descriptors are defined by XDM:
+
+* `xdm:identityContext`: allows a property in a schema to be used as an [Identity](https://github.com/adobe/xdm/blob/master/docs/reference/context/identity.schema.md), even if it does not conform to the Identity schema.
```
[the remark this answers is not in the document]
Comment (trieloff) [reply docs/descriptors.md]: @lrosenthol, @kstreeter – yes, that was my intention. I can remove the entire contentious section from the doc, we just have to remember putting them back in in the individual PRs.

**Comment by trieloff** [comment]
@lrosenthol Are we good to merge this so far? We can discuss the details of the various descriptors in separate PRs.

**Comment by lrosenthol** [comment]
Yes, go ahead and merge and we'll work out the details in separate PRs

**Pull request merged**